frontend/src/app/main/ui/workspace/sidebar/assets.cljs

```diff
@@ -200,22 +247,16 @@
             (st/emit! (dwu/start-undo-transaction))
             (apply st/emit!
                    (->> components
-                        (filter #(contains? selected-components (:id %)))
+                        (filter #(if multi-components?
```

Comment:
I think you should send just 1 event that changes all components instead of one per item. The issue with 1 event per component is that every event will change the objects and "force" a re-render. If you group them together and only change the state once you can have only 1 re-render. Also I think it's cleaner to move this filter to the `data` package.

Reply:
This may be interesting. But it requires that the selected assets info is in the global state, now it's in a state local to the components. I'll do this after all state is moved to global space (I've added a TODO note for me not to forget it).